util/internal/integrationtest/common/src/main/java/org/hibernate/search/util/impl/integrationtest/common/rule/BackendMock.java

```diff
@@ -421,9 +402,13 @@ public DocumentWorkCallListContext discarded() {
 			log.debugf( "Expecting %d works to be discarded", works.size() );
 			works.stream()
 					.map( work -> new DocumentWorkCall( indexName, DocumentWorkCall.WorkPhase.DISCARD, work ) )
-					.forEach( expectationConsumer );
+					.forEach( this::expect );
 			return this;
 		}
+
+		private void expect(DocumentWorkCall call) {
+			backendBehavior().getDocumentWorkCalls( call.documentKey() ).expectInOrder( call );
```

Comment:
Does still it make sense to have a method named `expectInOrder`?

Reply:
`expectInOrder`/`expectInAnyOrder` are defined in `CallQueue`, which is ised for for other purposes, in particular schema expectations where we need `expectInAnyOrder`.

So yes, it makes sense.

Reply:
OK +1 to merge it